src/Cli/Utils.cs

```diff
@@ -904,27 +904,37 @@ public static bool WriteJsonContentToFile(string file, string jsonContent)
         /// This method will check if DAB_ENVIRONMENT value is set.
         /// If yes, it will try to merge dab-config.json with dab-config.{DAB_ENVIRONMENT}.json
         /// </summary>
-        public static string MergeConfigsIfAvailable()
+        /// <returns>Returns the name of the merged Config if successful.</returns>
+        public static bool TryMergeConfigsIfAvailable(out string mergedFile)
         {
             string? environmentValue = Environment.GetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME);
+            mergedFile = string.Empty;
             if (!string.IsNullOrEmpty(environmentValue))
             {
-                string FileToBeOverridden = DefaultName;
-                string FileToBeOverriddenWith = $"{CONFIGFILE_NAME}.{environmentValue}{CONFIG_EXTENSION}";
-                string mergedFile = $"{CONFIGFILE_NAME}.{environmentValue}.merged{CONFIG_EXTENSION}";
+                string baseConfigFile = DefaultName;
+                string overrideConfigFile = RuntimeConfigPath.GetFileName(environmentValue, considerOverrides: false);
+                mergedFile = $"{CONFIGFILE_NAME}.{environmentValue}.merged{CONFIG_EXTENSION}";
 
-                if (DoesFileExistInCurrentDirectory(FileToBeOverridden) && DoesFileExistInCurrentDirectory(FileToBeOverriddenWith))
+                if (DoesFileExistInCurrentDirectory(baseConfigFile) && !string.IsNullOrEmpty(overrideConfigFile))
                 {
-                    string originalJson = File.ReadAllText(FileToBeOverridden);
-                    string newJson = File.ReadAllText(FileToBeOverriddenWith);
-                    string mergedJson = Merge(originalJson, newJson);
+                    try
+                    {
+                        string baseConfigJson = File.ReadAllText(baseConfigFile);
+                        string overrideConfigJson = File.ReadAllText(overrideConfigFile);
+                        string mergedConfigJson = Merge(baseConfigJson, overrideConfigJson);
 
-                    File.WriteAllText(mergedFile, mergedJson);
-                    return mergedFile;
+                        File.WriteAllText(mergedFile, mergedConfigJson);
+                        return true;
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, $"Failed to merge the config files.");
```

Comment:
Consideration for this and future PRs: Is logging the exception with a generic failure message useful and clear to the end user? Would a developer benefit from one line error messages which describe either a FileRead error (on file in path X), or file write error? The stack traces may not be the most actionable.